stdlib/public/runtime/CMakeLists.txt

```diff
@@ -75,8 +75,34 @@ set(swift_runtime_library_compile_flags ${swift_runtime_compile_flags})
 list(APPEND swift_runtime_library_compile_flags -DswiftCore_EXPORTS)
 list(APPEND swift_runtime_library_compile_flags -I${SWIFT_SOURCE_DIR}/stdlib/include/llvm/Support -I${SWIFT_SOURCE_DIR}/include)
 
-set(sdk "${SWIFT_HOST_VARIANT_SDK}")
+if(SWIFT_${SWIFT_HOST_VARIANT_SDK}_${SWIFT_HOST_VARIANT_ARCH}_ICU_STATICLIB)
+  set(libicu_i18n_a -licui18nswift)
+  set(libicu_uc_a -licuucswift)
+  set(libicu_data_a -licudataswift)
+else()
+  find_package(ICU REQUIRED COMPONENTS uc i18n)
+  get_filename_component(ICU_UC_LIBDIR "${ICU_UC_LIBRARIES}" DIRECTORY)
+  get_filename_component(ICU_I18N_LIBDIR "${ICU_I18N_LIBRARIES}" DIRECTORY)
```

Comment:
**This is code motion from further below.**

Not your fault, but this really is not what it should be doing.  The additional `find_package(ICU)` doesn't account for the custom build system.

In order to deal with the custom build system in Swift, the project added the custom variables to specify the library - `SWIFT_${SWIFT_HOST_VARIANT_SDK}_${SWIFT_HOST_VARIANT_ARCH}_ICU_{UC,I18N}`.  I think that we should continue to use this existing path rather than add all this additional complexity.    `SWIFT_${SWIFT_HOST_VARIANT_SDK}_${SWIFT_HOST_VARIANT_ARCH}_ICU_{I18N,UC}` should give you the path to the library.  Setting that appropriately should wire the path all the way through.  We should use that rather than the different paths for `-licu...` and the `find_package`.

Would be wonderful if you felt like doing that cleanup!

Reply:
Done. Added two commits in this PR to do this, since they are slightly non-trivial.